pkg/network/protocols/tls/gotls/testutil/gotls_server/gotls_server.go

```diff
@@ -0,0 +1,80 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// Package main is a simple client for the gotls_server.
+package main
+
+import (
+	"crypto/tls"
+	"flag"
+	"io"
+	"log"
+	"net"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/network/protocols/http/testutil"
+)
+
+func main() {
+	flag.Parse()
+	args := flag.Args()
+
+	file, err := os.CreateTemp("/tmp", "gotls_server-*.log")
+	if err != nil {
+		log.Fatalf("could not create log file: %s", err)
+	}
+	log.Default().SetOutput(file)
+
+	if len(args) < 1 {
+		log.Fatalf("usage: gotls_server <server_addr>")
+	}
+
+	addr := os.Args[1]
+
+	crtPath, keyPath, err := testutil.GetCertsPaths()
+	if err != nil {
+		log.Fatalf("Could not get certificates")
+	}
+
+	handler := func(w http.ResponseWriter, req *http.Request) {
+		statusCode := testutil.StatusFromPath(req.URL.Path)
+		if statusCode == 0 {
+			log.Printf("wrong request format %s", req.URL.Path)
```

Comment:
what will happen in such a case?
you'll continue to the writing the body, and you'll use the default status code

Reply:
Good catch. Though, this logic is straight from our HTTP server in our HTTP testutils, so we should fix it there as well.

I think this code path should, alongside the log, return a 400 (Bad Request) status code. This should make the client produce an error, caught by `require.NoError` and that will let a dev writing a test that its request is ill formed. Should I create a PR to fix that test util?